src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParserStateObjectManaged.cs

```diff
@@ -71,14 +71,34 @@ internal override void AssignPendingDNSInfo(string userProtocol, string DNSCache
 
         internal void ReadAsyncCallback(SNIPacket packet, uint error)
         {
-            ReadAsyncCallback(IntPtr.Zero, PacketHandle.FromManagedPacket(packet), error);
-            _sessionHandle?.ReturnPacket(packet);
+            SNIHandle sessionHandle = _sessionHandle;
+            if (sessionHandle != null)
+            {
+                ReadAsyncCallback(IntPtr.Zero, PacketHandle.FromManagedPacket(packet), error);
+                sessionHandle?.ReturnPacket(packet);
```

Comment:
NIT: null checking is not necessary.

Reply:
Sadly they are, they're there because I encountered some cases where the packets in-flight were causing errors on return because the sessionhandle had been nulled as the session was disposed due to an error elsewhere. In this case the items in the pool are going to be GC'ed anywhere so this drop is ok.

Reply:
I thought the same, but better being safe than sorry..